pytorch_lightning/core/saving.py

```diff
@@ -361,9 +361,10 @@ def save_hparams_to_yaml(config_yaml, hparams: Union[dict, Namespace]) -> None:
     if not os.path.isdir(os.path.dirname(config_yaml)):
         raise RuntimeError(f'Missing folder: {os.path.dirname(config_yaml)}.')
 
-    if Container is not None and isinstance(hparams, Container):
+    if Container is not None and (OmegaConf.is_config(hparams) or 
+                                  OmegaConf.is_config(v) for v in hparams.values() ):
         from omegaconf import OmegaConf
-        OmegaConf.save(hparams, config_yaml, resolve=True)
+        OmegaConf.save(OmegaConf.create(hparams), config_yaml, resolve=True)
```

Comment:
is the creation here a no-op, if it already is a config?

Reply:
checked with @omry, and it is not a no-op. Will add a filter.

Reply:
It works correctly but the semantics are of deepcopy. this is not free.

Reply:
Fixed to include logic to check this.